lib/regular-update.js

```diff
@@ -1,33 +1,44 @@
 'use strict';
 
-const request = require('request');
-
-// Map from URL to { timestamp: last fetch time, interval: in milliseconds,
-// data: data }.
+// Map from URL to { timestamp: last fetch time, data: data }.
 let regularUpdateCache = Object.create(null);
 
 // url: a string, scraper: a function that takes string data at that URL.
 // interval: number in milliseconds.
 // cb: a callback function that takes an error and data returned by the scraper.
-function regularUpdate(url, interval, scraper, cb, options) {
+function regularUpdate({
+  url,
+  intervalMillis,
+  json = true,
+  scraper = buffer => buffer,
+  options = {},
+  request = require('request'),
+}, cb) {
   const timestamp = Date.now();
-  const cache = regularUpdateCache[url];
-  if (cache != null &&
-      (timestamp - cache.timestamp) < interval) {
-    cb(null, regularUpdateCache[url].data);
+  const cached = regularUpdateCache[url];
+  if (cached != null &&
+      (timestamp - cached.timestamp) < intervalMillis) {
+    cb(null, cached.data);
     return;
   }
-  request(url, options || {}, function(err, res, buffer) {
+  request(url, options, (err, res, buffer) => {
     if (err != null) { cb(err); return; }
-    if (regularUpdateCache[url] == null) {
-      regularUpdateCache[url] = { timestamp: 0, data: 0 };
```

Comment:
These lines didn't make sense to me.

Reply:
Doesn't looks like it does anything to me.